app/controllers/alaveteli_pro/plans_controller.rb

```diff
@@ -4,7 +4,8 @@ class AlaveteliPro::PlansController < ApplicationController
   before_filter :authenticate
 
   def show
-    @plan = Stripe::Plan.retrieve(params[:id])
+    stripe_plan = Stripe::Plan.retrieve(params[:id])
+    @plan = AlaveteliPro::PlanWithTax.new(stripe_plan)
```

Comment:
this is going to be a bit clunky for reuse in the subscription code as we don't fetch the plan directly, it's accessed through `customer.subscription.plan`, having think

Reply:
it's not too bad inside a helper method (or similar) https://github.com/mysociety/alaveteli/blob/d3f7c14b9cff179c40bd2c80e39f04b3dce00f62/app/helpers/alaveteli_pro/account_helper.rb#L29-L31

Reply:
`@plan = AlaveteliPro::PlanWithTax.new(customer.subscription.plan)` seems fine to me

Reply:
Its not much different to e.g. needing a `plus_vat` helper…

```ruby
@plan = Stripe::Plan.retrieve(params[:id])
@price = add_vat(@plan.amount)

@plan = customer.subscription.plan
@price = add_vat(@plan.amount)
```

…still a 2-step process?

Reply:
sure, working with it but if I expand the pattern to make something similar to calculate discounts, I have 2 object instantiations, one to multiply by 1.2, the other to work out whether to take some money off

Reply:
You mean like?

```ruby
stripe_plan = Stripe::Plan.retrieve(params[:id])
@plan = PlanWithTax.new(PlanWithDiscount.new(plan, 20.0))
```
I quite like [this pattern](http://nithinbekal.com/posts/ruby-decorators/#applying-the-decorator-pattern).

Reply:
I don't think that's going to work here as I'm working with a `Stripe::Subscription` not a `Stripe::Plan` as discounts belong to `Subscriptions`

It looks like discount is applied net, then the tax is calculated, let me try to work out Stripe's discount structure a bit better...

Reply:
(that is nice, I want one - trying to work out how to get one)

Reply:
bizarrely...

```
stripe_sub = Stripe::Subscription.retrieve(params[:id])
@plan = PlanWithTax.new(PlanWithDiscount.new(subscription))
```